drivers: gpio: Add SDL simulated GPIO support

pdgendt opened the pull request:
This commit adds a driver to simulate GPIO state and interrupts from the keyboard when using SDL.

~I still need to make it build in CI. Would it make sense to extend `samples/subsys/display/lvgl` with example usage?~

pdgendt committed:
drivers: gpio: Add SDL emulated GPIO support

This commit adds a driver to simulate GPIO state and interrupts
using the keyboard when using SDL.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

cfriedt reviewed (commented):
It's quite possible that you could wrap around the existing `gpio_emul` driver for this and provide only the callbacks.

Did you already attempt that?

pdgendt committed:
samples: subsys: display: lvgl: Add button to reset counter

Connect the sw0 alias to reset the counter.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
samples: subsys: display: lvgl: Add native_posix overlay for SDL input

Enable SDL keyboard events using the zephyr,gpio-emul-sdl driver and
alias key to sw0.
Pressing 'r' on the keyboard resets the counter.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt commented:
> It's quite possible that you could wrap around the existing `gpio_emul` driver for this and provide only the callbacks.

Thanks @cfriedt for the suggestion, I've modified the driver as suggested.

mbolivar-nordic reviewed (dismissed):
@petejohanson not sure if this is relevant to you but FYI.

Binding looks fine.

petejohanson commented:
> @petejohanson not sure if this is relevant to you but FYI.
> 
> Binding looks fine.

Interesting! Perhaps, yes. We right now do have the ability to run as a posix process for some testing we do, this might make that a tad simpler.

No issues w/ the PR, other than maybe a request for some docs for posterity?

pdgendt commented:
> No issues w/ the PR, other than maybe a request for some docs for posterity?

Where would it make sense to add documentation? SDL is currently only mentioned in the `native_posix` [board support page](https://docs.zephyrproject.org/latest/boards/posix/native_posix/doc/index.html).

pdgendt committed:
drivers: gpio: pca953x: select I2C

Select I2C instead of depending on it, following upstream policy.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

mbolivar-nordic commented on samples/subsys/display/lvgl/src/main.c:
```diff
@@ -54,7 +101,7 @@ void main(void)
 			lv_label_set_text(count_label, count_str);
 		}
 		lv_task_handler();
-		k_sleep(K_MSEC(10));
 		++count;
+		k_sleep(K_MSEC(10));
```
Useless diff / unrelated change, it looks like.

pdgendt replied:
Not really, pressing the button resets the value immediately now, otherwise it would jump to 1 after a second

mbolivar-nordic replied:
Sorry I was unclear. If it's not useless diff, it's an unrelated change and the commit is doing 2 things.

pdgendt replied:
I would argument that it belongs with the commit `samples: subsys: display: lvgl: Add button to reset counter` as this introduces the new functionality.
Creating a new commit feels like a fixup.
